native-sql-engine/core/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala

```diff
@@ -375,6 +375,12 @@ case class ColumnarPostOverrides() extends Rule[SparkPlan] {
   var isSupportAdaptive: Boolean = true
 
   def replaceWithColumnarPlan(plan: SparkPlan): SparkPlan = plan match {
+    case RowToColumnarExec(child: BroadcastQueryStageExec) =>
+      logDebug(s"$child Due to a fallback of BHJ inserted into plan." +
+        s" See above override in BroadcastQueryStageExec")
+      val localBroadcastXchg = child.plan.asInstanceOf[BroadcastExchangeExec]
```

Comment:
may need to guard this logic, like check if child.isInstanceOf[xxx] first

Reply:
Fixed in new commit. Thanks!